hngprojects/hng_boilerplate_nestjs
# feat: Delete a user

**What does the PR do?**

This PR is about deleting a user

**How should this be manually tested?**

1. Login to a user account, and get the access token
2. Send a DELETE request to `/api/v1/users/:user_id`

Successful Response

On a successful user delete, the API should return a 200 SUCCESS status code.

```json
{
      "status": "success",
      "message": "Account Deleted Successfully",
      "data": {
            "user_id": "a9f4b3b0-4b3b-4b3b-4b3b-4b3b4b3b4b3b",
            "name": "John Doe",
            "phone_number":"+23467891234",
      }
}
```

Authentication Error:

When the current user does not have permission:

```json
{
          "status": "error",
          "message": "Bad Request",
          "status_code": "HttpStatus.BAD_REQUEST",
 }
```

Invalid UserID Error

If a wrong or invalid user_id is used:

```json
{
  "status": "Error",
  "message": "user_id is required",
  "status_code": 400
}
```

 **Checklist of what you have just done**
- [X ] Define API endpoint to soft delete a user.
- [X ]  Validate and handle authentication token.
- [ X]  Implement logic to soft delete user information from the database.
- [ X]  Handle success and error responses.
- [ X]  Test Successful response

**Test Screenshot**

![Screenshot from 2024-07-24 23-47-26](https://github.com/user-attachments/assets/f76f33cf-4135-4b8c-9c7c-f96bec346d24)

Reference Issue: [Link to Issue](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/167)
